### PR TITLE
Format and lint code, and establish standard tools for doing so

### PR DESCRIPTION
**Pull request by nstelter-slac**
to format and lint the code, the steps are as follows (all run from the root dir of the project):
_black --line-length 120 ._
_ruff --line-length=120 . --fix_
-fix remaining ruff issues that can't be auto-fixed, until _ruff --line-length=120_ runs without any output (indicates no remaining issues)

i'd say for now we keep this as a necessary step for developers, and maybe hard-enforce or automate in the pull-requests later.

all tests passing locally. also docs still building correctly.
also did a quick skim through the files and most changes are minor, but some to note:
-I added \_\_all\_\_ to the \_\_init\_\_.py files, this also stops from hitting ruff warnings for the files
-in a few spots I added "noqa" comments for lines to be skipped by the ruff linter, like those that do "import *" (ex: _pydm/PyQt/uic.py_). Seems like best way around odd cases where we need to break linting rules.

### CODE REVIEW

**Comment by klauer** [comment]
Looks like Python 3.7 in the test suite is failing.
On the photon side, we follow the numpy [NEP 29 deprecation policy](https://numpy.org/neps/nep-0029-deprecation_policy.html) which states that as of April this year, we only need to support Python 3.9+.
Is that sort of policy acceptable to PyDM's other users, I wonder?

**Comment by prjemian** [comment]
> Looks like Python 3.7 in the test suite is failing. On the photon side, we follow the numpy [NEP 29 deprecation policy](https://numpy.org/neps/nep-0029-deprecation_policy.html) which states that as of April this year, we only need to support Python 3.9+. Is that sort of policy acceptable to PyDM's other users, I wonder?

I can live with that as testing.  Might try to keep minimum Py3.8 as long as the code does not fail.

File: dev-requirements.txt [hunk 1]
```diff
@@ -5,3 +5,5 @@ pytest-cov
 pytest-timeout
 p4p
 pyca
+black
```
[the remark this answers is not in the document]
Comment (tangkong) [reply dev-requirements.txt:8]: CI job is here as well: https://github.com/pcdshub/pcds-ci-helpers/blob/master/.github/workflows/pre-commit.yml

**Review by jbellister-slac — COMMENTED** [review]
Thanks, this is looking great! I'm testing against some large displays, only issue so far was pmps not loading due to an import error where an `s` got dropped from `QtWebEngineWidgets` during a manual fix.

**Review by YektaY — APPROVED** [review]

**Review by jbellister-slac — APPROVED** [review]
All large displays I've tried run fine, new pre-commits looks good to me!

**Pull request merged**